Test fails when WCS support is removed

#2298 added the ability to turn off support for WCS.

I didn't properly test this, and at least one test will fail when WCS support is disabled because the test requires this support. So we need to add `@requires_wcs` to the test

```
=========================================================================== FAILURES ============================================================================
______________________________________________________________________ test_DataIMG_filter ______________________________________________________________________
[gw12] linux -- Python 3.13.5 /home/dburke/micromamba/envs/sherpa-3/bin/python3.13

caplog = <_pytest.logging.LogCaptureFixture object at 0x758df42d48a0>, clean_astro_ui = None

    @requires_fits
    def test_DataIMG_filter(caplog, clean_astro_ui):
        """Test filtering of DataIMG
    
        This is a slightly modified version of https://github.com/sherpa/sherpa/issues/1880
        """
        y = np.arange(6).reshape(2, 3) * 10 + 10
    
        sky = WCS("physical", "LINEAR", [100, 200], [1, 1], [10, 10])
        eqpos = WCS("world", "WCS", [30, 50], [100, 200], [-0.1, 0.1])
        data = ui.DataIMG.from_2d_array_with_wcs("faked", y=y,
                        sky=sky, eqpos=eqpos)
    
        ui.set_data(data)
>       ui.set_coord("physical")

sherpa/astro/ui/tests/test_astro_ui_data.py:44: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
<string>:1: in set_coord
    ???
sherpa/astro/ui/utils.py:7633: in set_coord
    self._get_img_data(id).set_coord(coord)
sherpa/astro/data.py:5778: in set_coord
    self.indep = func()
                 ^^^^^^
sherpa/astro/data.py:5740: in get_physical
    return self._get_coordsys("physical")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sherpa/astro/data.py:5734: in _get_coordsys
    return conv(x0, x1)
           ^^^^^^^^^^^^
sherpa/astro/data.py:5643: in _logical_to_physical
    x0, x1 = self.sky.apply(x0, x1)
             ^^^^^^^^^^^^^^^^^^^^^^
sherpa/astro/io/wcs.py:82: in apply
    x0, x1 = pix2world(self.type, x0, x1,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = ('LINEAR', array([1., 2., 1., 2., 1., 2.]), array([1., 1., 2., 2., 3., 3.]), array([1., 1.]), array([100., 200.]), array([10., 10.]), ...), kwargs = {}

    def pix2world(*args, **kwargs):
>       raise RuntimeError("No WCS support")
E       RuntimeError: No WCS support

sherpa/astro/io/wcs.py:41: RuntimeError
```

--- a/pytest.ini
+++ b/pytest.ini
@@ -82,5 +82,5 @@ doctest_subpackage_requires =
     docs/evaluation/examples.rst = group sherpa.astro.xspec._xspec
     # pycrates OR astropy would do, but the syntax does not allow to specify alternatives, so just pick one
     docs/evaluation/examples.rst = astropy
+    docs/fit/poisson.rst = group
     sherpa/astro/data.py = astropy
-

--- a/sherpa/astro/data.py
+++ b/sherpa/astro/data.py
@@ -159,6 +159,7 @@ from sherpa.astro.utils import arf_fold, rmf_fold, filter_resp, \
 
 __doctest_requires__ = {
     '.': ['sherpatest'],  # requirements for module-level doctest
+    'DataPHA.group_counts': ['group']
     }
 
 if TYPE_CHECKING:

--- a/sherpa/astro/ui/tests/test_astro_ui_data.py
+++ b/sherpa/astro/ui/tests/test_astro_ui_data.py
@@ -24,9 +24,10 @@ from tempfile import NamedTemporaryFile
 
 from sherpa.astro.io.wcs import WCS
 from sherpa.astro import ui
-from sherpa.utils.testing import requires_fits
+from sherpa.utils.testing import requires_fits, requires_wcs
 
 
+@requires_wcs
 @requires_fits
 def test_DataIMG_filter(caplog, clean_astro_ui):
     """Test filtering of DataIMG

--- a/sherpa/astro/ui/tests/test_serialize.py
+++ b/sherpa/astro/ui/tests/test_serialize.py
@@ -4285,6 +4285,7 @@ def test_store_default_id(defid):
 
 @requires_data
 @requires_fits
+@requires_group
 def test_copy_data_pha(make_data_path):
     """Can we copy a PHA dataset and track it?"""
 

